herbert_core/classes/instrument_classes/instrument/detectors/@IX_det_slab/effic.m

```diff
@@ -1,40 +1,44 @@
-function val = effic (obj, npath_in, varargin)
+function val = effic (obj, varargin)
 % Efficiency of a slab detector
 %
-%   >> val = effic (obj, npath, wvec)
-%   >> val = effic (obj, npath, ind, wvec)
+%   >> val = effic (obj, npath_in, wvec)
+%   >> val = effic (obj, ind, npath_in, wvec)
 %
 % Input:
 % ------
 %   obj         IX_det_slab object
 %
+%   ind         Indices of detectors for which to calculate. Scalar or array.
+%               Default: all detectors (i.e. ind = 1:ndet) as a row vector.
+%
 %   npath       Unit vectors along the neutron path in the detector coordinate
 %               frame for each detector. Vector length 3 or an array size [3,n]
 %               where n is the number of indices (see ind below). If a vector
-%               then npath is expanded internally to [3,n] array
-%
-%   ind         Indices of detectors for which to calculate. Scalar or array.
-%               Default: all detectors (i.e. ind = 1:ndet)
+%               then npath is expanded internally to [3,n] array.
 %
 %   wvec        Wavevector of absorbed neutrons (Ang^-1). Scalar or array.
-%
-% If both ind and wvec are arrays, then they must have the same number of elements
+%               If both ind and wvec are arrays, then they must have the same
+%               number of elements, but not necessarily the same shape.
 %
 %
 % Output:
 % -------
-%   val         Efficiency (in range 0 to 1) averaged over the detector area
+%   val         Efficiency (in range 0 to 1).
 %               The shape is whichever of ind or wvec is an array.
-%               If both ind and wvec are arrays, the shape is that of wvec
+%
+%               Note:
+%                 - if ind is a scalar, the calculation is performed for
+%                  that value at each of the values of wvec
+%                 - if wvec is a scalar, the calculation is performed for
+%                  that value at each of the values of ind
+%
+%               If both ind and wvec are arrays, the shape is that of wvec.
 
 
 % Original author: T.G.Perring
-%
-% $Revision:: 840 ($Date:: 2020-02-10 16:05:56 +0000 (Mon, 10 Feb 2020) $)
-
 
-[~,npath,ind,wvec] = parse_npath_ind_wvec_ (obj, npath_in, varargin{:});
-alf = macro_xs_thick (obj, npath, ind, wvec);
 
-val = effic_alf(alf);
+[~, ind, npath, wvec] = parse_ind_npath_wvec_ (obj, varargin{:});
+alf = macro_xs_thick (obj, ind, npath, wvec);
```

Comment:
Does "alf" refer to the instrument? Do we want a non-isis specific name of what this is?

Reply:
'alf' is nothing to do with the instrument